File: author-service/domain/Author.go
package domain

import (
	"github.com/gofrs/uuid/v5"
	"github.com/jackc/pgtype"
	"time"
)

type Author struct {
	ID        uuid.UUID   `json:"id"`
	Name      string      `json:"name"`
	Biography string      `json:"biography"`
	BirthDate pgtype.Date `json:"birthDate"`
	CreatedAt time.Time   `json:"createsAt"`
	UpdatedAt time.Time   `json:"updatedAt"`
}

File: author-service/src/main/kotlin/authorservice/model/Author.kt
package authorservice.model

import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.Lob
import org.hibernate.annotations.CreationTimestamp
import org.hibernate.annotations.UpdateTimestamp
import java.time.LocalDate
import java.time.LocalDateTime
import java.util.*

@Entity
data class Author(
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    val id: UUID? = null,
    var name: String,
    @Lob
    var biography: String,
    var birthDate: LocalDate,
    @CreationTimestamp
    val createdAt: LocalDateTime? = null,
    @UpdateTimestamp
    val updatedAt: LocalDateTime? = null
)
